File: 09_DataStructure_Algorithm/0518_day51/04_타겟넘버/타겟넘버.py
def solution(numbers, target):
    answer = 0
    
    def combi(idx, total):
        nonlocal answer
        
        if idx == len(numbers):
            if total == target:
                answer += 1
            return
        
        combi(idx + 1, total + numbers[idx])
        combi(idx + 1, total - numbers[idx])
    
    combi(0, 0)
    
    return answer

File: 09_DataStructure_Algorithm/0518_day51/04_타겟넘버/타겟넘버.java
class Solution {
    public int solution(int[] numbers, int target) {
        return dfs(numbers, target, 0, 0);
    }
    
    int dfs(int[] numbers, int target, int idx, int sum) {
        int count = 0;
        
        if (idx == numbers.length) {
            return target == sum ? 1 : 0;
        }
        
        count += dfs(numbers, target, idx + 1, sum + numbers[idx]);
        count += dfs(numbers, target, idx + 1, sum - numbers[idx]);
        
        return count;
    }
}